builtin/logical/database/path_roles.go

```diff
@@ -2,263 +2,606 @@ package database
+                                functionality. See the plugin's API page for more information on
+                                support and formatting for this parameter. `,
+				},
+				"rollback_statements": {
+					Type: framework.TypeStringSlice,
+					Description: `Specifies the database statements to be executed
+                                rollback a create operation in the event of an error. Not every
+                                plugin type will support this functionality. See the plugin's
+                                API page for more information on support and formatting for this
+                                parameter.`,
+				},
+
+				"default_ttl": {
+					Type:        framework.TypeDurationSecond,
+					Description: "Default ttl for role.",
+				},
+
+				"max_ttl": {
+					Type:        framework.TypeDurationSecond,
+					Description: "Maximum time a credential is valid for",
+				},
 			},
 
-			"db_name": {
-				Type:        framework.TypeString,
-				Description: "Name of the database this role acts on.",
-			},
-			"creation_statements": {
-				Type: framework.TypeStringSlice,
-				Description: `Specifies the database statements executed to
-				create and configure a user. See the plugin's API page for more
-				information on support and formatting for this parameter.`,
-			},
-			"revocation_statements": {
-				Type: framework.TypeStringSlice,
-				Description: `Specifies the database statements to be executed
-				to revoke a user. See the plugin's API page for more information
-				on support and formatting for this parameter.`,
-			},
-			"renew_statements": {
-				Type: framework.TypeStringSlice,
-				Description: `Specifies the database statements to be executed
-				to renew a user. Not every plugin type will support this
-				functionality. See the plugin's API page for more information on
-				support and formatting for this parameter. `,
-			},
-			"rollback_statements": {
-				Type: framework.TypeStringSlice,
-				Description: `Specifies the database statements to be executed
-				rollback a create operation in the event of an error. Not every
-				plugin type will support this functionality. See the plugin's
-				API page for more information on support and formatting for this
-				parameter.`,
+			ExistenceCheck: b.pathRoleExistenceCheck,
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ReadOperation:   b.pathRoleRead,
+				logical.CreateOperation: b.pathRoleCreateUpdate,
+				logical.UpdateOperation: b.pathRoleCreateUpdate,
+				logical.DeleteOperation: b.pathRoleDelete,
 			},
 
-			"default_ttl": {
-				Type:        framework.TypeDurationSecond,
-				Description: "Default ttl for role.",
+			HelpSynopsis:    pathRoleHelpSyn,
+			HelpDescription: pathRoleHelpDesc,
+		},
+
+		&framework.Path{
+			Pattern: "static-roles/" + framework.GenericNameRegex("name"),
+			Fields: map[string]*framework.FieldSchema{
```

Comment:
Since these are identical for many properties to those for `roles/` what about creating a function that takes a role type as a parameter and returns the correct map. This would allow the common elements (which are most of them) to stay consistent.

Reply:
I added a new method `fieldsForType` in 5b387f5a8 that does this (code docs added in 6babbfaa1)